modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java

```diff
@@ -884,6 +842,93 @@ public void testObjectFieldSettings() throws Exception {
 				objectFieldSettingValueException.getMessage());
 		}
 
+		try {
+			_objectFieldLocalService.addCustomObjectField(
+				null, TestPropsValues.getUserId(),
+				_listTypeDefinition.getListTypeDefinitionId(),
+				objectDefinition.getObjectDefinitionId(),
+				ObjectFieldConstants.BUSINESS_TYPE_PICKLIST,
+				ObjectFieldConstants.DB_TYPE_STRING, true, false, null,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				false, "picklist", true, true, Collections.emptyList());
+
+			Assert.fail();
+		}
+		catch (ObjectFieldSettingValueException.MissingRequiredValues
+					objectFieldSettingValueException) {
+
+			Assert.assertEquals(
+				"The settings defaultValue, defaultValueType are required " +
+					"for object field picklist",
+				objectFieldSettingValueException.getMessage());
+		}
+
+		// Invalid values
+
+		try {
+			_objectFieldLocalService.addCustomObjectField(
+				null, TestPropsValues.getUserId(),
+				_listTypeDefinition.getListTypeDefinitionId(),
+				objectDefinition.getObjectDefinitionId(),
+				ObjectFieldConstants.BUSINESS_TYPE_PICKLIST,
+				ObjectFieldConstants.DB_TYPE_STRING, true, false, null,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				false, "picklist", true, true,
+				Arrays.asList(
+					_createObjectFieldSetting(
+						ObjectFieldSettingConstants.NAME_DEFAULT_VALUE,
+						_listTypeEntryKey),
+					_createObjectFieldSetting(
+						ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE,
+						ObjectFieldSettingConstants.VALUE_EXPRESSION_BUILDER)));
+
+			Assert.fail();
+		}
+		catch (ObjectFieldSettingValueException.InvalidValue
+					objectFieldSettingValueException) {
```

Comment:
@devpedrotavares can you sort the exceptions alphabetically? `ObjectFieldSettingValueException.InvalidValue` before `ObjectFieldSettingValueException.MissingRequiredValues`